flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -3333,21 +3335,31 @@ SqlAlterModel SqlAlterModel() :
         <RENAME> <TO>
         newModelIdentifier = CompoundIdentifier()
         {
-            return new SqlAlterModel(
+            return new SqlAlterModelRename(
                         startPos.plus(getPos()),
                         modelIdentifier,
-                        newModelIdentifier,
-                        ifExists);
+                        ifExists,
```

Comment:
I am curious why you have reversed the order of these parameters?

Reply:
Good catch!